lib/python/rs274/glcanon.py

```diff
@@ -272,7 +287,7 @@ def tool_offset(self, xo, yo, zo, ao, bo, co, uo, vo, wo):
         self.xo = xo
         self.yo = yo
         self.zo = zo
-        self.so = ao
+        self.ao = ao
```

Comment:
99% Sure this was a typo that has always been an issue. But what is an "a" tool offset?

Reply:
Yes, that seems to be a typo. 
We can set tool offsets for all nine axis words (XYZABCUVW) in the tool table. I haven't used anything other then XYZ myself but maybe for a robot it's useful.

Reply:
On my gui, we go by either file size, or comment of `(DISABLE PREVIEW)`

```python
        if self.state.magic_comments["disable_preview"]:
            LOG.info("Preview Disabled by comments")
            self.ui.gcodegraphics.show_program = False
            return
        elif os.stat(file_name).st_size < self.max_graphics_file_size:
            self.ui.gcodegraphics.show_program = True
        else:
            self.ui.gcodegraphics.show_program = False
            LOG.warning(f"File To Large ({os.stat(file_name).st_size/1000000} mb), Preview disabled")

```

Reply:
Nice, looks quite useful.

Reply:
The alternative I would consider, but it would need adoption from axis, qtvcp and plasma is to make it a standalone toggle on/off.